packages/perf/test/index.spec.ts

```diff
@@ -0,0 +1,114 @@
+import type { TransportManager } from '@libp2p/interface-internal/transport-manager'
+
+const defaultInit: PerfServiceInit = {
+  protocolName: '/perf/1.0.0',
+  maxInboundStreams: 1 << 10,
+  maxOutboundStreams: 1 << 10,
+  timeout: 10000,
+  writeBlockSize: BigInt(64 << 10)
+}
+
+async function createComponents (): Promise<Components> {
+  const peerId = await createEd25519PeerId()
+
+  const events = new EventEmitter()
+
+  const components = defaultComponents({
+    peerId,
+    registrar: mockRegistrar(),
+    upgrader: mockUpgrader(),
+    datastore: new MemoryDatastore(),
+    transportManager: stubInterface<TransportManager>(),
+    connectionGater: mockConnectionGater(),
+    events
+  })
+
+  components.peerStore = new PersistentPeerStore(components)
+  components.connectionManager = new DefaultConnectionManager(components, {
+    minConnections: 50,
+    maxConnections: 1000,
+    autoDialInterval: 1000,
+    inboundUpgradeTimeout: 1000
+  })
+
+  return components
+}
+
+describe('perf', () => {
+  let localComponents: Components
+  let remoteComponents: Components
+
+  beforeEach(async () => {
+    localComponents = await createComponents()
+    remoteComponents = await createComponents()
+
+    await Promise.all([
+      start(localComponents),
+      start(remoteComponents)
+    ])
+  })
+
+  afterEach(async () => {
+    await Promise.all([
+      stop(localComponents),
+      stop(remoteComponents)
+    ])
+  })
+
+  it('should run perf', async () => {
+    const client = perfService(defaultInit)(localComponents)
+    const server = perfService(defaultInit)(remoteComponents)
+
+    await start(client)
+    await start(server)
+
+    // simulate connection between nodes
+    const [localToRemote, remoteToLocal] = connectionPair(localComponents, remoteComponents)
+    localComponents.events.safeDispatchEvent('connection:open', { detail: localToRemote })
+    remoteComponents.events.safeDispatchEvent('connection:open', { detail: remoteToLocal })
+
+    // Run Perf
+    await expect(client.perf(remoteComponents.peerId, 1024n, 1024n)).to.eventually.be.fulfilled()
+  })
+
+  it('should output benchmark', async () => {
+    const client = perfService(defaultInit)(localComponents)
+    const server = perfService(defaultInit)(remoteComponents)
+
+    await start(client)
+    await start(server)
+
+    // simulate connection between nodes
+    const [localToRemote, remoteToLocal] = connectionPair(localComponents, remoteComponents)
+    localComponents.events.safeDispatchEvent('connection:open', { detail: localToRemote })
+    remoteComponents.events.safeDispatchEvent('connection:open', { detail: remoteToLocal })
+
+    let downloadBandwidth = 0
+    let uploadBandwidth = 0
+
+    for (let i = 1; i < 5; i++) {
+      // Run Perf
+      downloadBandwidth += await client.measureDownloadBandwidth(remoteComponents.peerId, 10485760n)
+      uploadBandwidth += await client.measureUploadBandwidth(remoteComponents.peerId, 10485760n)
+    }
```

Comment:
```suggestion
    for (let i = 0; i < 5; i++) {
      // Run Perf
      downloadBandwidth += await client.measureDownloadBandwidth(remoteComponents.peerId, 10485760n)
      uploadBandwidth += await client.measureUploadBandwidth(remoteComponents.peerId, 10485760n)
    }
```

With my limited JS knowledge, this runs only `4` times otherwise.